catalog/app/containers/Admin/RolesAndPolicies/Policies.tsx

```diff
@@ -0,0 +1,735 @@
+import * as FF from 'final-form'
+import * as IO from 'io-ts'
+import * as R from 'ramda'
+import * as React from 'react'
+import * as RF from 'react-final-form'
+import * as urql from 'urql'
+import * as M from '@material-ui/core'
+
+import * as Notifications from 'containers/Notifications'
+import * as Model from 'model'
+import * as Dialogs from 'utils/Dialogs'
+import type FormSpec from 'utils/FormSpec'
+import assertNever from 'utils/assertNever'
+import { mkFormError, mapInputErrors } from 'utils/formTools'
+import * as Types from 'utils/types'
+import validate, * as validators from 'utils/validators'
+
+import * as Form from '../RFForm'
+import * as Table from '../Table'
+
+import AssociatedRoles from './AssociatedRoles'
+import BucketsPermissions from './BucketsPermissions'
+import { getArnLink } from './shared'
+
+import POLICIES_QUERY from './gql/Policies.generated'
+import POLICY_CREATE_MANAGED_MUTATION from './gql/PolicyCreateManaged.generated'
+import POLICY_CREATE_UNMANAGED_MUTATION from './gql/PolicyCreateUnmanaged.generated'
+import POLICY_UPDATE_MANAGED_MUTATION from './gql/PolicyUpdateManaged.generated'
+import POLICY_UPDATE_UNMANAGED_MUTATION from './gql/PolicyUpdateUnmanaged.generated'
+import POLICY_DELETE_MUTATION from './gql/PolicyDelete.generated'
+import { BucketPermissionSelectionFragment as BucketPermission } from './gql/BucketPermissionSelection.generated'
+import { PolicySelectionFragment as Policy } from './gql/PolicySelection.generated'
+
+const validateNonEmptyString: FF.FieldValidator<any> = validate(
+  'nonEmptyString',
+  validators.matches(/\S/),
+)
+
+const columns = [
+  {
+    id: 'title',
+    label: 'Title',
+    getValue: R.prop('title'),
+    props: { component: 'th', scope: 'row' },
+  },
+  {
+    id: 'source',
+    label: 'Source',
+    getValue: (p: Policy) => p.managed,
+    getDisplay: (value: boolean) =>
+      value ? (
+        <abbr title="This IAM policy is created and managed by Quilt">Quilt</abbr>
+      ) : (
+        <abbr title="This IAM policy is provided and managed by you or another administrator">
+          Custom
+        </abbr>
+      ),
+  },
+  {
+    id: 'buckets',
+    label: 'Buckets',
+    getValue: (p: Policy) => (p.managed ? p.permissions.length : null),
+    getDisplay: (_buckets: any, p: Policy) =>
+      p.managed ? (
+        <M.Tooltip
+          arrow
+          title={
+            p.permissions.length ? (
```

Comment:
I think it deserves to be a standalone component

Reply:
seems like more trouble than gain, since it's quite simple and not very reusable